apollo-api/src/main/java/com/apollographql/android/api/graphql/Field.java

```diff
@@ -89,7 +96,67 @@ public boolean optional() {
     return optional;
   }
 
-  public static enum Type {
+  public String cacheKey(Operation.Variables variables) {
+    if (arguments.isEmpty()) {
+      return fieldName();
+    }
+    return String.format("%s(%s)", fieldName(), orderIndependentKey(arguments, variables));
+  }
+
+  private String orderIndependentKey(Map<String, Object> objectMap, Operation.Variables variables) {
+    if (isArgumentValueVariableType(objectMap)) {
```

Comment:
maybe I'm wrong, but does this check means that we can't mix variables and arguments?
Do we handle when arguments have constants, plus variables?